chatmeleon-server/src/controllers/user-relationship-controller.ts

```diff
@@ -0,0 +1,125 @@
+import prisma from "../libs/prismadb";
+import { Request, Response } from "express";
+
+const createFriendRequestRelationship = async (req: Request, res: Response) => {
+  const { relatingUserId, relatedUserId } = req.body;
+  const relationship = await prisma.user_Relationship.createMany({
+    data: [
+      {
+        type: "sending_friend_request",
+        relatingUserId: relatingUserId,
+        relatedUserId: relatedUserId,
+      },
+      {
+        type: "pending_friend_request",
+        relatingUserId: relatedUserId,
+        relatedUserId: relatingUserId,
+      },
+    ],
+  });
+
+  res.json(relationship);
+};
+
+const createFriendRelationship = async (req: Request, res: Response) => {
+  const { relatingUserId, relatedUserId } = req.body;
+
+  const existingRelationship = await prisma.user_Relationship.findMany({
+    where: {
+      OR: [
+        {
+          relatingUserId: relatingUserId,
+          relatedUserId: relatedUserId,
+          type: "pending_friend_request",
+        },
+        {
+          relatingUserId: relatedUserId,
+          relatedUserId: relatingUserId,
+          type: "sending_friend_request",
+        },
+      ],
+    },
+  });
+
+  if (existingRelationship && existingRelationship.length === 2) {
+    const updateRelationship = await prisma.user_Relationship.updateMany({
+      where: {
+        OR: [
+          {
+            relatingUserId: relatingUserId,
+            relatedUserId: relatedUserId,
+            type: "pending_friend_request",
+          },
+          {
+            relatingUserId: relatedUserId,
+            relatedUserId: relatingUserId,
+            type: "sending_friend_request",
+          },
+        ],
+      },
+      data: {
+        type: "friend",
+      },
+    });
+
+    res.status(200).json({
+      message: "Friend relationship created successfully",
+      data: updateRelationship,
+    });
+  } else {
+    res.status(404).json({ message: "No pending friend request found" });
+  }
+};
+
+const deleteFriendRequestRelationship = async (req: Request, res: Response) => {
+  const { relatingUserId, relatedUserId } = req.body;
+
+  const existingRelationship = await prisma.user_Relationship.findMany({
+    where: {
+      OR: [
+        {
+          relatingUserId: relatingUserId,
+          relatedUserId: relatedUserId,
+          type: "pending_friend_request",
+        },
+        {
+          relatingUserId: relatedUserId,
+          relatedUserId: relatingUserId,
+          type: "sending_friend_request",
+        },
+      ],
+    },
+  });
+
+  if (existingRelationship && existingRelationship.length === 2) {
```

Comment:
I think this is redundant? I'm guessing deleting where no document is found in prisma wouldn't really be a problem. 2 db requests is a lil redundant

Reply:
can you explain more by what u mean

Reply:
This here `prisma.user_Relationship.findMany` the existing relation first, then if the relation exists, issue another command to `prisma.user_Relationship.deleteMany` the relation from the db. What I'm saying is I don't think `prisma.user_Relationship.findMany` first is needed since `prisma.user_Relationship.deleteMany` would probably do nothing (or report an error) if it cannot find any documents matching the `where` clause anyway (need confirmation on this though)

Reply:
yea got it, i was just being careful and accounted for all the error codes and responses but yea I'm lost on how to do these response codes and account for all the errors that we may have and maybe we need to talk this through again

Reply:
Yeah sure. One easy approach would just be reading the doc of prisma when implementing, to see what `deleteMany` would do, what would it throw and whatnot.